assignments/assignment02.rb

```diff
@@ -0,0 +1,157 @@
+# ========================================================================================
+# Assignment 2
+# ========================================================================================
+
+# ========================================================================================
+#  Problem 1 - `to_sentence`
+
+# implement method `to_sentence`
+
+# creates an english string from array
+
+def to_sentence(ary)
+
+  new_string = ""
+  input = ary.insert(-2, "and")
+  string = input.join(" ")
+  new_string << string
+
+
+  puts new_string
+  
+end
+
+# Your method should generate the following results:
+to_sentence []                       #=> ""
+to_sentence ["john"]                 #=> "john"
+to_sentence ["john", "paul"]         #=> "john and paul"
+to_sentence [1, "paul", 3, "ringo"]  #=> "1, paul, 3 and ringo"
+
+
+# ========================================================================================
+#  Problem 2 - `mean, median`
+
+# implement methods "mean", "median" on Array of numbers
+def mean(ary)
+  
+  mean = ary.inject(:+).to_f / ary.size
+
+  puts mean
+
+end
+
+def median(ary)
+  input = ary.sort
+  half = (input.length - 1) / 2
+  med = (input[half] + input[half.ceil]) / 2
+
+  puts med
+end
+
+# Your method should generate the following results:
+mean [1, 2, 3]    #=> 2.0
+mean [1, 1, 4]    #=> 2.0
+
+median [1, 2, 3]  #=> 2
+median [1, 1, 4]  #=> 1
+
+
+# ========================================================================================
+#  Problem 3 - `pluck`
+
+# implement method `pluck` on array of hashes
+def pluck(ary, key)
+  array = []
+
+  ary.each { |h| array << h[key] }
```

Comment:
you want to use map to create a new array, instead of using each and manually creating a new array.